src/libraries/System.IO.FileSystem/tests/Base/SymbolicLinks/BaseSymbolicLinks.FileSystem.cs

```diff
@@ -0,0 +1,462 @@
+        /// creates a file if the implementing class is for Directory or DirectoryInfo.</summary>
+        protected abstract void CreateFileOrDirectory(string path, bool createOpposite = false);
+        protected abstract void AssertIsCorrectTypeAndDirectoryAttribute(FileSystemInfo linkInfo);
+        protected abstract void AssertLinkExists(FileSystemInfo linkInfo);
+        /// <summary>Calls the actual public API for creating a symbolic link.</summary>
+        protected abstract FileSystemInfo CreateSymbolicLink(string path, string pathToTarget);
+        /// <summary>Calls the actual public API for resolving the symbolic link target.</summary>
+        protected abstract FileSystemInfo ResolveLinkTarget(string linkPath, bool returnFinalTarget = false);
+
+        /// <summary>
+        /// Verifies that FileSystemInfo.LinkTarget matches the specified expected path.
+        /// If the current platform is Windows and <paramref name="actual"/> is absolute, this method asserts that <paramref name="actual"/> starts with \\?\.
+        /// </summary>
+        protected static void AssertLinkTargetEquals(string expected, string actual)
+        {
+#if WINDOWS
+            if (Path.IsPathFullyQualified(actual))
+            {
+                actual.StartsWith(ExtendedPrefix);
+                expected = Path.Join(ExtendedPrefix, expected);
+            }
+
+            // Windows syscalls remove the redundant segments in the link target path.
+            // We will remove them from the expected path when testing Windows but keep them when testing Unix, which doesn't remove them.
+            int rootLength = PathInternal.GetRootLength(expected);
+            if (rootLength > 0)
+            {
+                expected = PathInternal.RemoveRelativeSegments(expected, rootLength);
+            }
+#endif
+            Assert.Equal(expected, actual);
+        }
+
+
+        /// <summary>
+        /// Asserts that the FullPath of the FileSystemInfo returned by ResolveLinkTarget() matches with the expected path of the file created.
+        /// Trims the Windows device prefix, in case there's any, before comparing.
+        /// </summary>
+        private static void AssertFullNameEquals(string expected, string actual)
+        {
+#if WINDOWS
+            if (PathInternal.IsExtended(actual))
+            {
+                Assert.StartsWith(ExtendedPrefix, actual);
+                actual = actual.Substring(4);
+            }
+#endif
+
+            Assert.Equal(expected, actual);
+        }
+
+        [Fact]
+        public void CreateSymbolicLink_NullPathToTarget()
+        {
+            Assert.Throws<ArgumentNullException>(() => CreateSymbolicLink(GetRandomFilePath(), pathToTarget: null));
+        }
+
+        [Theory]
+        [InlineData("")]
+        [InlineData("\0")]
+        public void CreateSymbolicLink_InvalidPathToTarget(string pathToTarget)
```

Comment:
is not allowing for empty paths our requirement or all OSes are nice and aligned in this case?

Reply:
I haven't tested what the behavior would be from the p/invokes if we pass an empty path, but I do know we have several public Path APIs that check for null or empty paths and for the invalid null character in the path. So this was done mainly to align the new symlink APIs to other existing APIs.